File: Data Structures & Algorithms/is-subsequence/submission-1.py
class Solution:
    def isSubsequence(self, s: str, t: str) -> bool:
        ls = 0
        rs = 0

        n = len(t)
        m = len(s)


        while ls < m and rs < n:
            if s[ls] == t[rs]:
                ls+=1
                rs+=1
            else:
                rs+=1
        
        if ls != m:
            return False
        return True

File: Data Structures & Algorithms/max-water-container/submission-1.cpp
class Solution {
public:
    int maxArea(vector<int>& heights) {
        // min (i,j) height * length of x (i->j)
        int maxarea = 0;
        int n = heights.size();
        int j = n-1, i=0;
        while(i<j){
            int area = min(heights[i], heights[j]) * (j-i);
            maxarea = max(maxarea, area);
            if(heights[i] > heights[j]){
                j--;
            }else{
                i++;
            }
        }

        return maxarea;
    }
};
